File: designer/types/change.ts
export type XChange = IChangeStandard | IChangeCustom;

export interface IChangeStandard {
	target: string;
	standard: string;
	custom?: never;
	value: string;
}

export interface IChangeCustom {
	target: string;
	standard?: never;
	custom: string;
	value: string;
}

File: api/BoningerWorks.TextAdventure.Json/Inputs/Change.cs
using System.Text.Json.Serialization;

namespace BoningerWorks.TextAdventure.Json.Inputs
{
	public class Change
	{
		[JsonPropertyName("target")] public string? Target { get; set; }
		[JsonPropertyName("standard")] public string? Standard { get; set; }
		[JsonPropertyName("custom")] public string? Custom { get; set; }
		[JsonPropertyName("value")] public string? Value { get; set; }
	}
}
